sphinx/writers/latex.py

```diff
@@ -1747,10 +1747,31 @@ def depart_citation_reference(self, node: Element) -> None:
     def visit_literal(self, node: Element) -> None:
         if self.in_title:
             self.body.append(r'\sphinxstyleliteralintitle{\sphinxupquote{')
+            return
         elif 'kbd' in node['classes']:
             self.body.append(r'\sphinxkeyboard{\sphinxupquote{')
-        else:
+            return
+        lang = node.get("language", None)
+        if 'code' not in node['classes'] or not lang:
             self.body.append(r'\sphinxcode{\sphinxupquote{')
+            return
+
+        def warner(msg):
+            logger.warning(msg, location=node)
+
+        highlight_args = dict(node.get("highlight_args", {}), nowrap=True)
+        opts = self.config.highlight_options.get(lang, {})
+        hlcode = self.highlighter.highlight_block(
+            node.astext(), lang, opts=opts,
+            location=node, **highlight_args
+        )
+        hlcode = hlcode.replace(r'\begin{Verbatim}[commandchars=\\\{\}]',
```

Comment:
Note: This macro can be changed via `highlight_args`. But as I commented above, the `code` role does not support it. So we don't need to care the output.

Additionally, it would be better to use `nowrap` option for the LaTeX formatter (refs: https://github.com/pygments/pygments/pull/1343) instead of str.replace(). But it's still not released yet. It would be better to mention the option as a comment here.

Reply:
Done.